jaxrs/src/main/java/se/fortnox/reactivewizard/jaxrs/JaxRsResultSerializerFactory.java

```diff
@@ -0,0 +1,92 @@
+package se.fortnox.reactivewizard.jaxrs;
+
+import com.fasterxml.jackson.core.type.TypeReference;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+import rx.functions.Func1;
+import se.fortnox.reactivewizard.json.JsonSerializerFactory;
+
+import javax.inject.Inject;
+import javax.ws.rs.core.MediaType;
+import java.nio.charset.Charset;
+import java.nio.charset.StandardCharsets;
+import java.util.Collections;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+import static reactor.core.publisher.Flux.just;
+
+public class JaxRsResultSerializerFactory {
+    private static final Class<byte[]>   BYTEARRAY_TYPE = byte[].class;
+    private static final Charset         charset        = StandardCharsets.UTF_8;
+
+    private final JsonSerializerFactory jsonSerializerFactory;
+
+    @Inject
+    public JaxRsResultSerializerFactory(JsonSerializerFactory jsonSerializerFactory) {
+        this.jsonSerializerFactory = jsonSerializerFactory;
+    }
+
+    /**
+     * Creates a non streaming result serializer
+     * @param type Content-Type
+     * @param dataCls Return type of the resource method
+     * @param isFlux True if it is a flux, false otherwise
+     * @param <T> Type of elements emitted by the publisher
+     * @return a function that serializes the elements emitted by a publisher to a Flux of byte arrays.
+     */
+    public <T> Func1<Flux<T>, Flux<byte[]>> createNonStreamingSerializer(String type, Class<T> dataCls, boolean isFlux) {
```

Comment:
Can't we use an ordinary java util functions mostly everywhere?

Reply:
Since it was an Func1 before my changes I thought I'd leave it as is. That said, I do agree that it should be a java util function rather than an rx function.